### DISCUSSION TITLE
Pull HLS Streams

### DISCUSSION THREAD

**Discussion by irlkitcom** [OvenMediaLabs/OvenMediaEngine#1580] [Help]
Is it possible to pull HLS streams?

**Comment by bchah** [comment]
Out of OME, yes - into OME, no. OME pulls RTSP or MPEG-2 TS.

Now, there may be some way to set this up using [Scheduled Channel](https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/scheduledchannel-api) but it would require you to somehow mux the HLS chunks into flattened files and add them to a Scheduled Channel Program in sequence. But this would be rather complex and unlikely to run with stability.

**Reply by irlkitcom** [reply]
This is available in other streaming servers. Would be a great addition to OME.

**Reply by irlkitcom** [reply]
Also, according to the documentation, only RTSP pulling is supported. Is there any documentation on MPEG-2 TS pulling?

**Reply by bchah** [reply]
I may be mistaken about TS

**Reply by getroot** [reply]
I think it would be a good idea to pull HLS with ffmpeg and push it to OME over rtmp or srt.
HLS is a protocol for viewers, and there are many much better protocols created for ingest than that, so I don't see a reason to use HLS for ingest.

**Reply by irlkitcom** [reply]
I could do that, but it's an additional point of failure as well as, I'd like to use UnusedStreamDeletionTimeout to conserve bandwidth. This functionality is available in Nimble, AMS, and Wowza.

**Reply by irlkitcom** [reply]
What sponsorship tier gets custom features? 😅

**Reply by getroot** [reply]
To redistribute HLS streams, just a reverse proxy with a cache function will be sufficient. Do you want to pull HLS and record/re-encode/stream it to another protocol?

Sometimes a sponsorship increases the priority of a particular task or registers it as a new task. But that's not guaranteed. And we've been so busy these days that big features like HLS Pull won't be developed quickly even with sponsorship.

**Reply by irlkitcom** [reply]
I'm wanting to transcode it into multiple qualities.

**Reply by privatesam** [reply]
How can OME pull MPEG-2 TS?   In the documentation it seems to imply it can only be pushed?
Features

    Ingest

        Push: WebRTC, WHIP, SRT, RTMP, MPEG-2 TS

        Pull: RTSP
